### ISSUE TITLE
panic: gee_array_list_real_get: assertion failed: (index < _size)

### ISSUE THREAD

**Issue by stevenroose** [dino/dino#1165]
I have a problem that looks a bit like https://github.com/dino/dino/issues/137 (but that was fixed in 2017).

Second time in a row, after running for a while, Dino crashes with this:
```
ERROR:arraylist.c:736:gee_array_list_real_get: assertion failed: (index < _size)
Bail out! ERROR:arraylist.c:736:gee_array_list_real_get: assertion failed: (index < _size)
[1]    1679779 abort (core dumped)  env G_MESSAGES_DEBUG=libdino,OMEMO ~/software/dino/build/dino
```

Last `OMEMO-DEBUG` logs before are:
```
(dino:1679779): OMEMO-DEBUG: 17:51:38.992: stream_module.vala:112: Asking for bundle for <redacted>
(dino:1679779): OMEMO-DEBUG: 17:51:38.992: manager.vala:221: new bundles 4/4 for <redacted>
```

I was running commit 4e9957de, which was the one-bit tip at the time of writing this issue.

**Comment by stevenroose** [comment]
I just had this again before I pulled in master. Now rebuilt (839be0e316bf756ddccb9eba17b75c974440093c) and will report if I ever see it again.

**Comment by fiaxh** [comment]
Thanks for reporting, but unfortunately it's rather impossible to debug this without a stack trace.

If your system is still similar to how it [was](https://github.com/dino/dino/issues/934#issuecomment-775255756) in early 2021, it probably generated a core dump. Please generate a stack trace from that core dump by entering `coredumpctl info [pid]` and then `bt`. You can find out the pid by looking though the list of stack traces with `coredumpctl list -r`.

**Comment by fiaxh** [comment]
I actually came across such an error myself today and was able to fix it (c70201080b9bf432b9b7be2a88fd3cb6efa781b8). I hope that there aren't too many of this exact error in our code, so I'll assume that it's the same one. If you experience this crash again, please leave a message and I'll reopen the issue.
